plugins/woocommerce/includes/wccom-site/installation/installation-steps/class-wc-wccom-site-installation-step-activate-product.php

```diff
@@ -79,6 +79,11 @@ private function activate_plugin( $product_id ) {
 			throw new Installer_Error( Installer_Error_Codes::UNKNOWN_FILENAME );
 		}
 
+		// If the plugin is already active, make sure we call the registration hook.
+		if ( is_plugin_active( $filename ) ) {
+			WC_Helper::activated_plugin( $filename );
+		}
+
 		$result = activate_plugin( $filename );
```

Comment:
We won't need to activate the plugin, if you find it already active based on previous check.

Reply:
I think you mean moving the next set of lines to the else clause? I tried to keep changes minimal especially since `activate_plugin` is almost a no-op in this case.

Almost because it [checks plugin requirements again](https://github.com/WordPress/wordpress-develop/blob/6.5/src/wp-admin/includes/plugin.php#L659) which might return a `WP_Error`.